src/application/application.ts

```diff
@@ -25,13 +27,35 @@ class Application {
             this.api.ExecuteTask(req, handleAPIResponse(resolve, reject))
         })
     }
+
+    executeTasKAndWaitFirstResult(req: ExecuteTaskRequest): Promise<ResultData> {
+        return new Promise<ResultData>((resolve, reject) => {
+            const id = uuidv4()
+            const stream = this.listenResult({ serviceID: req.serviceID, tagFilters: [id] })
+                .on('error', (err) => { reject(err) })
+                .on('metadata', (metadata) => {
+                    const err = checkStreamReady(metadata)
+                    if (err) return
```

Comment:
On err it should reject

Reply:
Yes, but there is a little problem with that. I tested this with a scenario where the _serviceID_ doesn't exist in the Core while calling the _executeTaskAndWaitFirstResult()_ func. 

The problem is, if _serviceID_ is wrong, Core will never send a status metadata because it early returns with the error. In this case _checkStreamReady()_ will fail and we shouldn't raise this error with _reject()_. Instead, we should raise the error we'll get from the _error_ event which says that service cannot be found in the database.

The catch here is, by the nature, _metadata_ event is emitted first before the _error_ event. And we cannot know that if _error_ event will be emitted after _metadata_. Thus, we don't know we should reject with the status check error we have inside the metadata handler or wait for the _error_ event.

One way of solving this is, I think, we should always assume that Core will send a status ready metadata when the stream is successfully initialized. And if not, we shouldn't expect the status and if there is no status sent, we shouldn't reject inside the metadata handler and simply early return.

So, this will require us to handle empty status & wrong status errors differently returned by the _checkStreamReady()_.